### ISSUE TITLE
Should revert changes if publish fails

### ISSUE THREAD

**Issue by tgandrews** [atom/apm#477]
Actions:
0. Package at version `1.0.0`.
1. I forgot to perform a `git pull` before an `apm publish minor`. 
2. This caused the publish to fail.
3. Ran `git pull`
4. Ran `apm publish minor`

Actual
APM published version 1.2.

Expected
APM to publish version 1.1

**Comment by 50Wliu** [comment]
This is because `apm publish` successfully changed the package version number in package.json and created a new tag, but failed to push it and update atom.io.  Therefore the next time you ran `apm publish` it would increment the version number again, and then publish that version to atom.io.

If you still want it to publish 1.1 you would have to revert the commit that bumped the version number, and also delete the tag.

**Comment by 50Wliu** [comment]
Oh, sorry, I didn't see that you *wanted* it to automatically revert.

**Comment by tgandrews** [comment]
Yes, from the command line you just retrieve an error - so I assumed that the publish had failed. It may just require different output when an error is encountered.

**Comment by surdu** [comment]
As another scenario where this is hurtful, I ended up bumping two major version because of this issue and the fact that login failed due to Two Factor Authentication.
